dao/src/test/scala/za/co/absa/enceladus/dao/rest/CrossHostApiCallerSuite.scala

```diff
@@ -27,12 +27,55 @@ class CrossHostApiCallerSuite extends BaseTestSuite {
     Mockito.reset(restClient)
   }
 
+  "CrossHostApiCaller" should {
+    "hold correct url and try count pairs" when {
+      "the url and try counts are fo same size" in {
+        val crossHostApiCaller = CrossHostApiCaller(Vector("a", "b", "c", "d"), Vector(1, 2, 3, 4), 3)
+        crossHostApiCaller.nextBaseUrl() should be("a", 1)
+        crossHostApiCaller.nextBaseUrl() should be("b", 2)
+        crossHostApiCaller.nextBaseUrl() should be("c", 3)
+        crossHostApiCaller.nextBaseUrl() should be("d", 4)
+        crossHostApiCaller.nextBaseUrl() should be("a", 1)
```

Comment:
```suggestion
        val crossHostApiCaller = CrossHostApiCaller(Vector("a", "b", "c", "d"), Vector(1, 2, 3, 4), 2)
        crossHostApiCaller.nextBaseUrl() should be("d", 4)
        crossHostApiCaller.nextBaseUrl() should be("a", 1)
        crossHostApiCaller.nextBaseUrl() should be("b", 2)
        crossHostApiCaller.nextBaseUrl() should be("c", 3)
        crossHostApiCaller.nextBaseUrl() should be("d", 4)
```

the last param of `CrossHostApiCaller.apply(..., startWith: Int)`, the `startWith`, seems to be implemented correctly, but the tests do not really demonstrate the case where the `nextBaseUrl()` can thus return URL from the middle of the list as its purpose is.